### PR TITLE
Using requests for SOAP calls

### PR DESCRIPTION
**Pull request by clintonb**
This switch from requests to urllib2 should resolve our issues with SSL verification.

ECOM-3099

### CODE REVIEW

**Comment by clintonb** [comment]
@rlucioni please review
@feanil FYI

Our investigation got a few things wrong:
1. We were using `openssl s_client` incorrectly. We should have been specifying `CAfile` and/or `CApath` arguments: http://stackoverflow.com/questions/25156180/how-to-list-certificates-trusted-by-openssl.
2. The client was telling us exactly what we needed to fix; but, I missed it since I've never used the client before. The second line of output tells us exactly which certificate in the chain is not trusted. However, even when specifying the a single file or an entire directory, verification still fails. I have tried with certificates from CyberSource and those from Entrust.

```
[clintonb2] clintonb@clintonb2 i-8e44f338:~/ics2ws_certs$ openssl s_client -showcerts -connect ics2ws.ic3.com:443
CONNECTED(00000003)
depth=2 C = US, O = "Entrust, Inc.", OU = www.entrust.net/CPS is incorporated by reference, OU = "(c) 2006 Entrust, Inc.", CN = Entrust Root Certification Authority
verify error:num=20:unable to get local issuer certificate
verify return:0
---
Certificate chain
 0 s:/C=US/ST=California/L=Mountain View/1.3.6.1.4.1.311.60.2.1.3=US/1.3.6.1.4.1.311.60.2.1.2=Delaware/O=Cybersource Corporation/businessCategory=Private Organization/serialNumber=2838921/CN=ics2ws.ic3.com
   i:/C=US/O=Entrust, Inc./OU=www.entrust.net/rpa is incorporated by reference/OU=(c) 2009 Entrust, Inc./CN=Entrust Certification Authority - L1E
-----BEGIN CERTIFICATE-----
MIIFfjCCBGagAwIBAgIETCEJxTANBgkqhkiG9w0BAQUFADCBsTELMAkGA1UEBhMC
VVMxFjAUBgNVBAoTDUVudHJ1c3QsIEluYy4xOTA3BgNVBAsTMHd3dy5lbnRydXN0
Lm5ldC9ycGEgaXMgaW5jb3Jwb3JhdGVkIGJ5IHJlZmVyZW5jZTEfMB0GA1UECxMW
KGMpIDIwMDkgRW50cnVzdCwgSW5jLjEuMCwGA1UEAxMlRW50cnVzdCBDZXJ0aWZp
Y2F0aW9uIEF1dGhvcml0eSAtIEwxRTAeFw0xNDAxMjMyMDQ5MjNaFw0xNjA0MDgw
MjIyMDlaMIHWMQswCQYDVQQGEwJVUzETMBEGA1UECBMKQ2FsaWZvcm5pYTEWMBQG
A1UEBxMNTW91bnRhaW4gVmlldzETMBEGCysGAQQBgjc8AgEDEwJVUzEZMBcGCysG
AQQBgjc8AgECEwhEZWxhd2FyZTEgMB4GA1UEChMXQ3liZXJzb3VyY2UgQ29ycG9y
YXRpb24xHTAbBgNVBA8TFFByaXZhdGUgT3JnYW5pemF0aW9uMRAwDgYDVQQFEwcy
ODM4OTIxMRcwFQYDVQQDEw5pY3Myd3MuaWMzLmNvbTCCASIwDQYJKoZIhvcNAQEB
BQADggEPADCCAQoCggEBAIkJfD7S3uEYFPAKbH6Q0QIMEiY5WP5iL2H0cEIb2der
JwaSdcwdk9NV7Pgks+W1DSAPLDT/77LNAj6NTLvhpjYkBNRdOKUKzk5GLY/JdDGP
qbeSxYPKLoADXJkL6z9Sk8K9MRun9qAJSRgX77PqWc2VNDxH6WFgLhP3B3dc9uAB
s3lHQ6mLDfQvApFQ0PWWRzO0K1lRCuYxOQstbZq1Di5NEg3pYkSjty89fmD4P+0Y
0KWZ5G2aINlGi60OVZLkSr5Rml/bIvgoSM7zRvFE/3LQHO9Y237fpXtjHSbqbn5e
dsF/3/cB/S4ul4XHIAAHU/0WztOU40H+5GYxAYeVYHMCAwEAAaOCAXUwggFxMAsG
A1UdDwQEAwIFoDAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwZQYIKwYB
BQUHAQEEWTBXMCMGCCsGAQUFBzABhhdodHRwOi8vb2NzcC5lbnRydXN0Lm5ldDAw
BggrBgEFBQcwAoYkaHR0cDovL2FpYS5lbnRydXN0Lm5ldC9sMWUtY2hhaW4uY2Vy
MDMGA1UdHwQsMCowKKAmoCSGImh0dHA6Ly9jcmwuZW50cnVzdC5uZXQvbGV2ZWwx
ZS5jcmwwQQYDVR0gBDowODA2BgpghkgBhvpsCgECMCgwJgYIKwYBBQUHAgEWGmh0
dHA6Ly93d3cuZW50cnVzdC5uZXQvcnBhMBkGA1UdEQQSMBCCDmljczJ3cy5pYzMu
Y29tMB8GA1UdIwQYMBaAFFtBirLEQ8G9v8hUQVWd4Jat/7mhMB0GA1UdDgQWBBQE
Udl5F6E8ZDy2gV7zSPEdfGEm2TAJBgNVHRMEAjAAMA0GCSqGSIb3DQEBBQUAA4IB
AQAgn8+T2LdsgTHNaXE7zVQS9Oqgr8mGjwUzPr6pTTO9xLeNyzL0/V4joRb4XCIz
glBVGesASp5kRC3K6bpe5JkQvkydRfcC/LeZQDInHbsETBJLD/XRApOhVim3ZO0L
yVomz8PRlDNQoGck8DeGTnRVujWo+vUBrq0rccqB21T/S+f6YsJI+3nOfTaEWGbQ
kWCVTRg0a4dODw5T/Yvo4sMFyDzHrUfqBs1cWX/xH0buhEHkQVnkvtojzsn5QRSg
tGupzZl08j8JulFJpDMBavPIi7SeAizycq+bBY+IjSJKbrVSJ2TcJo5IGMCah2LJ
Fs1sbfh2mQ8fY7RidOiAoBKE
-----END CERTIFICATE-----
 1 s:/C=US/O=Entrust, Inc./OU=www.entrust.net/rpa is incorporated by reference/OU=(c) 2009 Entrust, Inc./CN=Entrust Certification Authority - L1E
   i:/C=US/O=Entrust, Inc./OU=www.entrust.net/CPS is incorporated by reference/OU=(c) 2006 Entrust, Inc./CN=Entrust Root Certification Authority
-----BEGIN CERTIFICATE-----
MIIFCjCCA/KgAwIBAgIERWua3DANBgkqhkiG9w0BAQUFADCBsDELMAkGA1UEBhMC
VVMxFjAUBgNVBAoTDUVudHJ1c3QsIEluYy4xOTA3BgNVBAsTMHd3dy5lbnRydXN0
Lm5ldC9DUFMgaXMgaW5jb3Jwb3JhdGVkIGJ5IHJlZmVyZW5jZTEfMB0GA1UECxMW
KGMpIDIwMDYgRW50cnVzdCwgSW5jLjEtMCsGA1UEAxMkRW50cnVzdCBSb290IENl
cnRpZmljYXRpb24gQXV0aG9yaXR5MB4XDTA5MTIxMDIwNTU0M1oXDTE5MTIxMDIx
MjU0M1owgbExCzAJBgNVBAYTAlVTMRYwFAYDVQQKEw1FbnRydXN0LCBJbmMuMTkw
NwYDVQQLEzB3d3cuZW50cnVzdC5uZXQvcnBhIGlzIGluY29ycG9yYXRlZCBieSBy
ZWZlcmVuY2UxHzAdBgNVBAsTFihjKSAyMDA5IEVudHJ1c3QsIEluYy4xLjAsBgNV
BAMTJUVudHJ1c3QgQ2VydGlmaWNhdGlvbiBBdXRob3JpdHkgLSBMMUUwggEiMA0G
CSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQC2WwRUd90OJGbcKqHbgMxdx1/9UhZY
2l+UBqm4trljDEcgguzHlU6LuHdSaj21h6nW4cx05abIwNRWT40u1gg+DExDPvBB
k15G7znn2WUqDHZQJ71bDTMzB+D3oqmc4REzrWb80ix6qqNzFr6ThXUP1zeM+iO3
ZPjjTG7tswW94jbbfN52RNqCcna2bv+UodCG9xDNSlqLsHWMZlKATkhMSYOmQNd3
gRNNXnJ+SEYiqg/iPmWUOOFycf5KcQm6NX9ViT2B1bgoARB3NloQhdK9YIQrSWGU
DN5MQGoqxHlghCSCMmlKmEviVhC6A0VRINPP2o5UG0W2erqXmlrYxtFfAgMBAAGj
ggEnMIIBIzAOBgNVHQ8BAf8EBAMCAQYwDwYDVR0TAQH/BAUwAwEB/zAzBggrBgEF
BQcBAQQnMCUwIwYIKwYBBQUHMAGGF2h0dHA6Ly9vY3NwLmVudHJ1c3QubmV0MDMG
A1UdHwQsMCowKKAmoCSGImh0dHA6Ly9jcmwuZW50cnVzdC5uZXQvcm9vdGNhMS5j
cmwwOwYDVR0gBDQwMjAwBgRVHSAAMCgwJgYIKwYBBQUHAgEWGmh0dHA6Ly93d3cu
ZW50cnVzdC5uZXQvQ1BTMB0GA1UdDgQWBBRbQYqyxEPBvb/IVEFVneCWrf+5oTAf
BgNVHSMEGDAWgBRokORnpKZTgMeGZqTx90tD+4S9bTAZBgkqhkiG9n0HQQAEDDAK
GwRWNy4xAwIAgTANBgkqhkiG9w0BAQUFAAOCAQEAsjvSnsG8O0i23NhaGGZTw701
DUhCLDUB2BCi4uONLLqmAxHta7FJy1/N7GCzutQC62FPTn7435BfTtOQAhxS2hIA
L5tx2gQSFMGQgy4o0hBAEYsmLeuZVVRvYI7Fgx3Aoz/VihQ5ahsN79NadznPabS9
aW9PeNOhhqObt9f7qi3w+iah+WcsiEulNNWD+0zxW3AiZhubWU9NzpjbQaT+GqPr
OOb58TkCnUa2ycKePoK2H5/KSqixBl8QNDv92nusM07tprdL85H1nAsRktwTasjV
8TttlmsB5CNMscHg0hIhnynUrZU9pvfnMsV1twtX2KT5wOzsMjMMTa7oCNXsqg==
-----END CERTIFICATE-----
 2 s:/C=US/O=Entrust, Inc./OU=www.entrust.net/CPS is incorporated by reference/OU=(c) 2006 Entrust, Inc./CN=Entrust Root Certification Authority
   i:/C=US/O=Entrust.net/OU=www.entrust.net/CPS incorp. by ref. (limits liab.)/OU=(c) 1999 Entrust.net Limited/CN=Entrust.net Secure Server Certification Authority
-----BEGIN CERTIFICATE-----
MIIEmzCCBASgAwIBAgIEQoctTDANBgkqhkiG9w0BAQUFADCBwzELMAkGA1UEBhMC
VVMxFDASBgNVBAoTC0VudHJ1c3QubmV0MTswOQYDVQQLEzJ3d3cuZW50cnVzdC5u
ZXQvQ1BTIGluY29ycC4gYnkgcmVmLiAobGltaXRzIGxpYWIuKTElMCMGA1UECxMc
KGMpIDE5OTkgRW50cnVzdC5uZXQgTGltaXRlZDE6MDgGA1UEAxMxRW50cnVzdC5u
ZXQgU2VjdXJlIFNlcnZlciBDZXJ0aWZpY2F0aW9uIEF1dGhvcml0eTAeFw0wNzAx
MDUxOTIwMzlaFw0xNzAxMDUxOTUwMzlaMIGwMQswCQYDVQQGEwJVUzEWMBQGA1UE
ChMNRW50cnVzdCwgSW5jLjE5MDcGA1UECxMwd3d3LmVudHJ1c3QubmV0L0NQUyBp
cyBpbmNvcnBvcmF0ZWQgYnkgcmVmZXJlbmNlMR8wHQYDVQQLExYoYykgMjAwNiBF
bnRydXN0LCBJbmMuMS0wKwYDVQQDEyRFbnRydXN0IFJvb3QgQ2VydGlmaWNhdGlv
biBBdXRob3JpdHkwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQC2lbZD
QvrGbSpvSN+UTDlXBe7DeRFBaDbt7P6aAY+hOCj89xBGZi5NHhqxGk7G0cCViLDJ
/zGLMwPbt4N7PiCEXu2yViin+OC5QHE3xctHDpcqaMAilWIV20fZ9dAr/4JLya0+
3kzbkIBQPwmKhADsMAo9GM37/SpZmiOVFyxFnh9uQ3ltDFyY/kinxSNHXF79buce
tPZoRdGGg1uiio2x4ymA/iVxiK2+vI+sUpZLqlGN5BMxGehOTZ/brLNq1bw5VHHK
enp/kN19HYDZgbtZJsIR/uaT4veA5GX7NDcOKYBwTa84hi6ef1evnheu6xzLKCFf
thzY56IEIvnT2tjLAgMBAAGjggEnMIIBIzAOBgNVHQ8BAf8EBAMCAQYwDwYDVR0T
AQH/BAUwAwEB/zAzBggrBgEFBQcBAQQnMCUwIwYIKwYBBQUHMAGGF2h0dHA6Ly9v
Y3NwLmVudHJ1c3QubmV0MDMGA1UdHwQsMCowKKAmoCSGImh0dHA6Ly9jcmwuZW50
cnVzdC5uZXQvc2VydmVyMS5jcmwwOwYDVR0gBDQwMjAwBgRVHSAAMCgwJgYIKwYB
BQUHAgEWGmh0dHA6Ly93d3cuZW50cnVzdC5uZXQvQ1BTMB0GA1UdDgQWBBRokORn
pKZTgMeGZqTx90tD+4S9bTAfBgNVHSMEGDAWgBTwF2ITVT2z/woAa/tQhJfz7WLQ
GjAZBgkqhkiG9n0HQQAEDDAKGwRWNy4xAwIAgTANBgkqhkiG9w0BAQUFAAOBgQAM
sIR8LRP+mj2/GAWVPSBIoxaBhxVQFaSIjZ9g1Dpv6y1uOoakqdLBnYl6CBykLbNH
jg9kSm9mA4M/TzSUNqopbYuNAiIrjM13pXCVhpHRtr9SvjNqa5n5b+ESvgTLM7/1
EhpORLpbFk0wufO0dM5u8mhWWN3Yof1UBfQjkYXJ+Q==
-----END CERTIFICATE-----
---
Server certificate
subject=/C=US/ST=California/L=Mountain View/1.3.6.1.4.1.311.60.2.1.3=US/1.3.6.1.4.1.311.60.2.1.2=Delaware/O=Cybersource Corporation/businessCategory=Private Organization/serialNumber=2838921/CN=ics2ws.ic3.com
issuer=/C=US/O=Entrust, Inc./OU=www.entrust.net/rpa is incorporated by reference/OU=(c) 2009 Entrust, Inc./CN=Entrust Certification Authority - L1E
---
No client certificate CA names sent
---
SSL handshake has read 4094 bytes and written 655 bytes
---
New, TLSv1/SSLv3, Cipher is AES256-SHA256
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : AES256-SHA256
    Session-ID: 8B9795875C3C422D4955E49156FF5C926CD2545378CF55A3116190B5DD70C41A
    Session-ID-ctx: 
    Master-Key: A69E2EDF6B80FBB8AA5893C98E4BE522D8376349056CAB77F0A4CABDDBF54616A34B8AC97A8B6204FB754431A064C30A
    Key-Arg   : None
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    Start Time: 1449539059
    Timeout   : 300 (sec)
    Verify return code: 20 (unable to get local issuer certificate)
---
```

I still have no clue how to make `openssl` or `curl` work with these certificates. I have tried on a couple sandboxes as well as my personal server to no avail. This bug may be the cause, but I am not certain: https://bugs.launchpad.net/ubuntu/+source/openssl/+bug/396818.

Nevertheless, @e0d's earlier question about `urllib2` got me wondering why `requests` works for these calls. It turns out `requests` [bundles its own CA certificates](http://docs.python-requests.org/en/latest/user/advanced/#ca-certificates), including an Entrust certificate. Furthermore, `suds` can be adapted to use `requests` instead of `urllib2`: http://stackoverflow.com/questions/6277027/suds-over-https-with-cert. Hence this PR, which should resolve the issue for refunds.

**Comment by feanil** [comment]
Tested in production. :+1:

File: ecommerce/extensions/payment/transport.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    """
+    HTTP transport utilizing requests library.
+
+    This class uses requests, instead of urllib2, to make HTTP requests. This allows us to properly
+    verify SSL certificates. This has been adapted from
+    http://stackoverflow.com/questions/6277027/suds-over-https-with-cert.
+    """
+    def open(self, request):
+        """
+        Fetches the WSDL using cert.
```
Comment (clintonb) [inline ecommerce/extensions/payment/transport.py]: I am updating this and the docstring below to be more accurate.

File: ecommerce/extensions/payment/tests/test_transport.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                               API_URL,
+                               body=body,
+                               content_type=CONTENT_TYPE,
+                               forcing_headers={'date': 'never'})
+
+        transport = RequestsTransport()
+        response = transport.send(request)
+
+        self.assertEqual(response.code, 200)
+        print type(response.headers)
```
Comment (rlucioni) [inline ecommerce/extensions/payment/tests/test_transport.py]: Remove.

**Comment by rlucioni** [comment]
@clintonb looks good aside from one stray print statement. :+1: once tests pass.

**Pull request merged**